router/batchrouter/asyncdestinationmanager/snowpipestreaming/types.go

```diff
@@ -127,6 +131,15 @@ type (
 		destination  *backendconfig.DestinationT
 		api
 	}
+
+	authzBackoff struct {
+		// time at which the an attempt was made to create a resource but it failed likely due to permission issues.
+		lastestErrorTime time.Time
+		// If lastAttemptedTime is not zero, then the next attempt to create a SF connection will be made after backoffDuration.
+		// This approach prevents repeatedly activating the warehouse even though the permission issue remains unresolved.
+		backoffDuration        time.Duration
+		initialBackoffDuration time.Duration
```

Comment:
Can we use `github.com/cenkalti/backoff/v4` instead of implementing this logic?

Reply:
Replaced the custom implementation with the library